h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -447,18 +449,32 @@ public void modelFailed(Model failedModel) {
 
     /** All visited hyper params permutations, including the current one. */
     private List<int[]> _visitedPermutations = new ArrayList<>();
+    private int _numberOfNonSkippedPermutations = 0;
```

Comment:
since you plan to apply this concept to other walkers as well, I think this is not the best solution, I think ideally this functionality would be a wrapper over any BaseWalker class (or an extracted interface) and as such would be easily unit testable in separation

Reply:
@honzasterba As I mentioned above, separation is not efficient. Also I don't see any issues with testing current design solution - see tests. You can still test filter functions separately if you like.